container/container_contract.go

```diff
@@ -182,13 +219,46 @@ func Put(container []byte, signature interop.Signature, publicKey interop.Public
 
 	addContainer(ctx, containerID, ownerID, cnr)
 
+	if name != "" {
+		if needRegister {
```

Comment:
`if name != "" && needRegister` will be easier to read with one less indent, probably

Reply:
There is another logic after the inner `if`, I would prefer to leave it as it is.